components/dashboard/src/teams/TeamSettings.tsx

```diff
@@ -397,3 +405,86 @@ function OrgDefaultWorkspaceImageModal(props: OrgDefaultWorkspaceImageModalProps
         </Modal>
     );
 }
+
+interface WorkspaceClassOptionsProps {
+    settings: OrganizationSettings | undefined;
+    disabled: boolean;
+}
+const WorkspaceClassOptions = (props: WorkspaceClassOptionsProps) => {
+    const [errorMsg, setErrorMsg] = useState("");
+    const [selectedValue, setSelectedValue] = useState(props.settings?.allowedWorkspaceClasses ?? []);
+    const updateTeamSettings = useUpdateOrgSettingsMutation();
+    const { data: classes, isError, isLoading } = useWorkspaceClasses();
+
+    const { toast } = useToast();
+    const handleUpdateTeamSettings = useCallback(
+        async (classes: string[]) => {
+            try {
+                await updateTeamSettings.mutateAsync(
+                    {
+                        ...props.settings,
+                        allowedWorkspaceClasses: classes,
+                    },
+                    {
+                        onSuccess: () => {
+                            toast({ message: "Available workspace classes updated." });
+                        },
+                    },
+                );
+            } catch (error) {
+                if (!ErrorCode.isUserError(error["code"])) {
+                    console.error(error);
+                }
+                setErrorMsg(error.message);
+            }
+        },
+        [updateTeamSettings, props, toast],
+    );
+
+    if (isError || !classes) {
+        return <div>Something went wrong</div>;
+    }
+
+    if (isLoading) {
+        return <LoadingState />;
+    }
+
+    return (
+        <>
+            {errorMsg.length > 0 && (
+                <Alert type="error" closable={true} className="mb-2 max-w-xl rounded-md">
+                    {errorMsg}
+                </Alert>
+            )}
+            {classes.map((wsClass) => (
+                <CheckboxInputField
+                    key={wsClass.id}
+                    id={wsClass.id}
+                    label={wsClass.displayName}
+                    hint={wsClass.description}
+                    checked={selectedValue.length === 0 || selectedValue.includes(wsClass.id)}
+                    onChange={(checked) => {
+                        const oldVal = selectedValue;
+                        let tmpSelectedValue = selectedValue;
+                        if (tmpSelectedValue.length === 0) {
+                            tmpSelectedValue = classes.map((e) => e.id);
+                        }
+                        const newVal = checked
+                            ? [...tmpSelectedValue, wsClass.id]
+                            : tmpSelectedValue.filter((e) => e !== wsClass.id);
+                        if (newVal.length === 0) {
+                            setErrorMsg("Should have one workspace class selected at least.");
```

Comment:
Rather than trying to track this state of if a class is selected we can derived the value and avoid state. For example, in our render method we can do something like:

```tsx
const noClassesSelected = selectedClasses.length === 0;
```

Then render an error message for that case. In regards to that, we can wrap the list of `<CheckboxInputFIeld>` components with `<CheckboxListField>` and that provides a property to render errors associated to those fields. For example:

```tsx
<CheckboxListField label="..." error="...">
  <CheckboxInputFIeld .../>
  <CheckboxInputFIeld .../>
</CheckboxListField>
```